File: cs_version3/cs_version3/Hr_manager.cs
/***********************************************************************
 * Module:  Hr_manager.cs
 * Author:  Anton
 * Purpose: Definition of the Class Hr_manager
 ***********************************************************************/

using System;
using System.Collections.Generic;

public class Hr_manager : Worker
{
    public Hr_manager() : base() 
    {
        //name = "n";
        Console.WriteLine("HR-manager was created (default)");
    }
	public Hr_manager(string n) : base(n)
    {
        //name = n;
        Console.WriteLine("HR-manager was created (inicialisation)");
    }
	public Hr_manager(Hr_manager sHr_manager)
    {
        //name = sHr_manager.name;
        Console.WriteLine("HR-manager was created (copy)");
    }
   public List<string> prepareVacancyList(int max, int cur)
    {
       List<string> list = new List<string>(1);
	for (int i = 0; i < (max - cur); i++)
	{
		string vac;
		vac = Console.ReadLine();
		list.Add(vac);
	}
	return list;
    }

	public void chooseCandidates(List<string> list)
   {
       for (int i = 0; i < candidate.Count; i++)
       {
           for (int j = 0; j < list.Count; j++)
           {
               if (candidate[i].resume.getCompetence() == list[j])
               {
                   candidate[i].setState("review");
                   break;
               }
           }
       }
   }

   public List<Candidate> candidate = new List<Candidate>(1);
   
   /// <pdGenerated>default getter</pdGenerated>
   //public System.Collections.ArrayList GetCandidate()
   //{
   //   if (candidate == null)
   //      candidate = new System.Collections.ArrayList();
   //   return candidate;
   //}
   
   ///// <pdGenerated>default setter</pdGenerated>
   //public void SetCandidate(System.Collections.ArrayList newCandidate)
   //{
   //   RemoveAllCandidate();
   //   foreach (Candidate oCandidate in newCandidate)
   //      AddCandidate(oCandidate);
   //}
   
   ///// <pdGenerated>default Add</pdGenerated>
   //public void AddCandidate(Candidate newCandidate)
   //{
   //   if (newCandidate == null)
   //      return;
   //   if (this.candidate == null)
   //      this.candidate = new System.Collections.ArrayList();
   //   if (!this.candidate.Contains(newCandidate))
   //      this.candidate.Add(newCandidate);
   //}
   
   ///// <pdGenerated>default Remove</pdGenerated>
   //public void RemoveCandidate(Candidate oldCandidate)
   //{
   //   if (oldCandidate == null)
   //      return;
   //   if (this.candidate != null)
   //      if (this.candidate.Contains(oldCandidate))
   //         this.candidate.Remove(oldCandidate);
   //}
   
   ///// <pdGenerated>default removeAll</pdGenerated>
   //public void RemoveAllCandidate()
   //{
   //   if (candidate != null)
   //      candidate.Clear();
   //}

   //private string name;

}

File: version_7/version_7/Hr_manager.cpp
/***********************************************************************
* Module:  Hr_manager.cpp
* Author:  Anton
* Modified: Monday, November 7, 2016 6:57:48 PM
* Purpose: Implementation of the class Hr_manager
***********************************************************************/
#include <vector>
#include <string>
#include "Candidate.h"
#include "Hr_manager.h"
#include "Technical_manager.h"
#include "Error.h"

////////////////////////////////////////////////////////////////////////
// Name:       Hr_manager::prepareVacancyList()
// Purpose:    Implementation of Hr_manager::prepareVacancyList()
// Return:     void
////////////////////////////////////////////////////////////////////////

std::vector<std::string> Hr_manager::prepareVacancyList(int max, int cur)//can be better
{
	std::vector<std::string> list;
	for (int i = 0; i < (max - cur); i++)
	{
		std::string vac;
		std::cin >> vac;
		list.push_back(vac);
	}
	return list;
	// TODO : implement
}

////////////////////////////////////////////////////////////////////////
// Name:       Hr_manager::chooseCandidates()
// Purpose:    Implementation of Hr_manager::chooseCandidates()
// Return:     void
////////////////////////////////////////////////////////////////////////

void Hr_manager::chooseCandidates(std::vector<std::string> list)
{
	for (int i = 0; i < candidate.size(); i++)
	{
		for (int j = 0; j < list.size(); j++)
		{
			if (candidate[i]->resume.getCompetence() == list[j])
			{
				candidate[i]->setState("review");
				break;
			}
		}
	}
	// TODO : implement
}

std::ostream &operator<< (std::ostream &stream, Hr_manager hrmanager)
{
	stream << "name: " << hrmanager.name << "\n";
	return stream;
}

Hr_manager Hr_manager::operator + (Candidate &right)
{
	std::string s = " ";
	try{
		if (this->candidate.size() >= 5)
		{
			throw Error(2, &s);
		}
		this->candidate.push_back(&right);
	}
	catch(Error e){
		e.err(&s);
	}
	return *this;
}

////////////////////////////////////////////////////////////////////////
// Name:       Hr_manager::get()
// Purpose:    Implementation of Hr_manager::get()
// Return:     void
////////////////////////////////////////////////////////////////////////

//std::string  Hr_manager::getName(void)
//{
//	return name;
//}

////////////////////////////////////////////////////////////////////////
// Name:       Hr_manager::set()
// Purpose:    Implementation of Hr_manager::set()
// Return:     void
////////////////////////////////////////////////////////////////////////

//void Hr_manager::setName(std::string n)
//{
//	name = n;
//}